redis/asyncio/client.py

```diff
@@ -723,13 +728,18 @@ async def __aenter__(self):
         return self
 
     async def __aexit__(self, exc_type, exc_value, traceback):
-        await self.reset()
+        await self.aclose()
 
     def __del__(self):
         if self.connection:
             self.connection.clear_connect_callbacks()
 
-    async def reset(self):
```

Comment:
This now removes a public method named reset - something we've shied away from doing. WDYT about keeping it defined, and also marking as deprecated?  At the very least we should mark as breakingchange and then not merge to 5.1 otherwise.

Reply:
sure, we can keep it around.  If you can provide me with a guideline on how to mark methods and attributes as deprecated, and how we can schedule those for future removal, that would be awesome.

Reply:
@kristjanvalur  see my comment bellow about how to mark function as deprecated. for attributes, you can't use the decorator, so you just need to import warnings and warn deprecated if the user try to use this attribute, like we did [here](https://github.com/redis/redis-py/blob/ba186d210f4f1e5d16fb27a1ee0afcb930f16a8d/redis/client.py#L203).